patches/server/0010-Adventure.patch

```diff
@@ -2605,6 +2605,26 @@ index 5a541e14caccaca97759879fd081becd0e8c6af2..13859822feebb65f14973aa688a54fcb
          buf.writeComponent(this.header);
          buf.writeComponent(this.footer);
      }
+diff --git a/src/main/java/net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java b/src/main/java/net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java
+index a5578cbce2d1d39ba3315e53425d5323e6823a65..29533d8881d99ee77945ecc91e4df48558d08d63 100644
+--- a/src/main/java/net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java
++++ b/src/main/java/net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java
+@@ -17,7 +17,14 @@ public class ClientboundLoginDisconnectPacket implements Packet<ClientLoginPacke
+ 
+     @Override
+     public void write(FriendlyByteBuf buf) {
+-        buf.writeUtf(Component.Serializer.toJson(this.reason));
++        // Paper start - Adventure
++        if (this.reason instanceof io.papermc.paper.adventure.AdventureComponent adv) {
++            buf.writeUtf(io.papermc.paper.adventure.PaperAdventure.asJsonString(adv.adventure$component(), buf.adventure$locale), FriendlyByteBuf.MAX_COMPONENT_STRING_LENGTH);
```

Comment:
I would generally like to note that afaict, that locale field will always be null

Reply:
is that still the case with the config phase being added? I dont have an overview anymore at what point that event is fired

Reply:
In the login phase locale field is always null, but rendered with the default locale.
Would it be better to change the locale argument to null? (since the result will be the same anyway)

Reply:
> is that still the case with the config phase being added?

Yes, because this packet only exists within login

I was earlier thinking that i'd be somewhat inclined to explictly pass null and leave a note as to why, headache comes into the expectations of events and such using this, maybe be inclined to add a javadoc there, but 🤷

Reply:
Okay, I added Javadoc in API and changed the locale argument to null.